src/backend/controllers/marketController.js

```diff
@@ -214,6 +214,56 @@ exports.addMarketSale = async (req: Request, res: Response) => {
     }
 }
 
+// Gets all Marketplace Sales of the user that are active
+exports.getUsersMarketSales = async (req: Request, res: Response) => {
+    // Validation
+    const errors = validationResult(req);
+
+    if (!errors.isEmpty()) {
+        // Return 400 for a bad request
+        return res
+            .status(400)
+            .json({ errors: errors.array() });
+    }
+
+    // Deconstruct userId from parameter
+    const { userId } = req.params;
+
+    try {
+        // Check if valid user
+        const user = await User.findById(userId);
+        if (!user) {
+            return res
+                .status(400)
+                .json({ msg: 'User does not exist' });
+        }
+
+		// Get list of sales from DB that belong to logged in user
+		const salesListOfTanks = await MarketSale.find({ sellerId: userId, itemType: { $eq: 'tank' }}).populate('itemId', 'tankName');
+		const salesListOfComponents = await MarketSale.find({ sellerId: userId , itemType: { $eq: 'component' }});
+		const salesList = salesListOfTanks.concat(salesListOfComponents);
+		/*for(let eachSale in salesList) {
+			if(salesListOfComponents[eachSale].sellerId == userId) {
+				salesListWithUserOnly.push(salesListOfComponents[eachSale]);
+			}
+		}*/
+        if (!salesList) {
+            return res
+                .status(400)
+                .json({ msg: 'Unable to get list of Market Sales.' });
+        }
+
+        // Return list of sales
+		console.log('Retrieved User\'s Market Sales List.');
```

Comment:
Nit: If you use double quotes you don't have to escape the single quotes. That is the main time where double quotes are recommended in JS.